test/core.jl

```diff
@@ -7198,3 +7198,11 @@ struct NFANode34126
 end
 
 @test repr(NFANode34126()) == "$NFANode34126(Tuple{Nothing,$NFANode34126}[])"
+
+@testset "issue #34544/35367" begin
+    # Test these evals dont throw an error
+    eval(Expr(:call, :eval, Expr(:quote, Expr(:module, true, :bar, Expr(:block)))))
+    eval(Expr(:module, true, :bar, Expr(:block)))
+    eval(Expr(:quote, Expr(:module, true, :bar, Expr(:quote))))
+    eval(Expr(:call, :eval, Expr(:quote, Expr(:module, true, :bar, Expr(:quote)))))
```

Comment:
I don't think this should be valid; the 3rd argument to `module` has to be a `block` expression. That can be checked at the top of `jl_eval_module_expr`.